src/main/java/org/variantsync/diffdetective/diff/difftree/DiffNode.java

```diff
@@ -968,17 +968,17 @@ public boolean isElse() {
     }
 
     /**
-     * Returns true if this node is a root node (i.e., it has {@link NodeType#ROOT}.
+     * Returns true if this node is a root node (has no parents).
      */
     public boolean isRoot() {
-        return this.nodeType.equals(NodeType.ROOT);
+        return getBeforeParent() == null && getAfterParent() == null;
     }
 
     /**
      * Returns {@link NodeType#isAnnotation()} for this node's {@link DiffNode#nodeType}.
      */
     public boolean isAnnotation() {
-        return this.nodeType.isAnnotation();
+        return this.nodeType.isAnnotation() && !this.isRoot();
```

Comment:
Please remove this extra check. The methods `isAnnotation` and `isArtifact` should partition any DiffTree.